content/blog/automated-video-transcoding-with-openfaas.md

```diff
@@ -0,0 +1,109 @@
+---
+title: "Automated Video Transcoding with OpenFaaS"
+date: 2018-06-09
+linktitle: Automated Video Transcoding
+draft: false
+---
+
+# Purpose
+
+I have a very large movie collection and we use Plex to keep it organized and easy to manage. That means a lot of storage space is needed for movie files, especially when they are Blue-ray. To save space on disk while maintaining the good quality I built this transcode pipeline to automate the process of adding a new movie to the library. You can find the repo [here](https://github.com/cpitkin/openfaas-transcode)
+
+## Uploader
+
+The companion uploader executable can be found [here](https://github.com/cpitkin/video-upload/)
+
+# Architecture
+
+![architecuture diagram](/img/tp.png)
+
+## Considerations
+
+- Transcoding is a very CPU intensive task. Regardless of how you configure the setup, it is going to take significant CPU and disk i/o resources. That being said you can run this setup on pretty much any hardware but the more resources you give the transcode container the faster it will run. This setup has been tested using an Ubuntu Linux box with a 6 core 3.5 GHz AMD processor.
```

Comment:
Mention how much RAM you have. If possible the AMD processor type. (Run NodeInfo..)